store/tikv/kv.go

```diff
@@ -198,6 +208,44 @@ func (s *KVStore) BeginWithExactStaleness(txnScope string, prevSec uint64) (*KVT
 	return txn, nil
 }
 
+// BeginWithMinStartTS begins transaction with the least startTS
+func (s *KVStore) BeginWithMinStartTS(txnScope string, minStartTS uint64) (*KVTxn, error) {
+	stores := make([]*Store, 0)
+	allStores := s.regionCache.getStoresByType(tikvrpc.TiKV)
+	if txnScope != oracle.GlobalTxnScope {
+		for _, store := range allStores {
+			if store.IsLabelsMatch([]*metapb.StoreLabel{
+				{
+					Key:   "zone",
+					Value: txnScope,
+				},
+			}) {
+				stores = append(stores, store)
+			}
+		}
+	} else {
+		stores = allStores
+	}
+	resolveTS := s.getMinResolveTSByStores(stores)
```

Comment:
What if `stores` is empty? The resolveTS will be MaxUint64.

Reply:
updated, If the stores is empty, we will directly return 0 in order to let txn use minStartTS as startTS.